### PR TITLE
[Doc] update max-autotune for CPU

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134986

The current doc for `max-autotune` is applicable only for GPU. This PR adds the corresponding content for CPU.

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

**Comment by chunyuan-w** [comment]
Hi @malfet could you help review this PR when you get time? We hope to land it before the PyTorch 2.5 branch cut.

**Comment by jgong5** [comment]
Hi @malfet @jansel Does the change look good to you? Thanks.

**Comment by malfet** [comment]
Hi @jgong5 I think it's a bit too technical, and should probably contain a phrase "enables performance tuning on support backends" and than add an explanation, that for cpu it's done for matmul templates and for GPUs for matmuls and convolutions.

File: torch/__init__.py [hunk 1]
```diff
@@ -2382,8 +2382,9 @@ def compile(
           There are other circumstances where CUDA graphs are not applicable; use TORCH_LOG=perf_hints
           to debug.
 
-        - "max-autotune" is a mode that leverages Triton based matrix multiplications and convolutions
-          It enables CUDA graphs by default.
+        - "max-autotune" is a mode that leverages CPP template-based matrix multiplications on CPU
+          and Triton based matrix multiplications and convolutions on GPU.
```
Comment (malfet) [inline torch/__init__.py:2386]:
```suggestion
        - "max-autotune" is a mode that leverages Triton or  template based matrix multiplications 
          on supported devices and Triton based convolutions on GPU.
```

**Review by malfet — APPROVED** [review]

File: torch/__init__.py (hunk 1 above)
Comment (chunyuan-w) [reply torch/__init__.py]: Thanks for the suggestion and fix.
Seems there's an issue of the change committed directly when using ghstack. Let me re-submit the change you proposed via ghstack.
image

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**